pkg/volume/volume_linux.go

```diff
@@ -36,60 +39,192 @@ const (
 // SetVolumeOwnership modifies the given volume to be owned by
 // fsGroup, and sets SetGid so that newly created files are owned by
 // fsGroup. If fsGroup is nil nothing is done.
-func SetVolumeOwnership(mounter Mounter, fsGroup *int64) error {
-
+func SetVolumeOwnership(mounter Mounter, fsGroup *int64, fsGroupChangePolicy *v1.PodFSGroupChangePolicy) error {
 	if fsGroup == nil {
 		return nil
 	}
 
+	fsGroupPolicyEnabled := utilfeature.DefaultFeatureGate.Enabled(features.ConfigurableFSGroupPolicy)
+
 	klog.Warningf("Setting volume ownership for %s and fsGroup set. If the volume has a lot of files then setting volume ownership could be slow, see https://github.com/kubernetes/kubernetes/issues/69699", mounter.GetPath())
 
-	return filepath.Walk(mounter.GetPath(), func(path string, info os.FileInfo, err error) error {
+	// This code exists for legacy purposes, so as old behaviour is entirely preserved when feature gate is disabled
+	// TODO: remove this when ConfigurableFSGroupPolicy turns beta.
```

Comment:
remove this when it's turns *GA*

Reply:
are we sure GA? My plan is to *at least* use new faster loop to use recursively chown files even if feature is beta. The newer loop does not sorts files before iterating on them, so should be bit quicker.

Reply:
alright renamed beta to GA.  we can discuss if new behaviour should be enabled or not in a future PR.